### DISCUSSION TITLE
Cannot connect using diffie-hellman-group key exchange algorithms

### DISCUSSION THREAD

**Discussion by tsanna82** [drakkan/sftpgo#845] [General]
Hi,
I'm trying to connect an AWS instance to an old SFTP server which is using deprecated key exchange algorithms.
The response is: 
```
ssh: handshake failed: error waiting for session: ssh: no common algorithm for key exchange; client offered: [curve25519-sha256 curve25519-sha256@libssh.org ecdh-sha2-nistp256 ecdh-sha2-nistp384 ecdh-sha2-nistp521 diffie-hellman-group14-sha256 ext-info-c], server offered: [diffie-hellman-group14-sha1 diffie-hellman-group-exchange-sha1 diffie-hellman-group1-sha1]
```

I've tried to edit sftpgo.json as follows:
```
"kex_algorithms": [
"diffie-hellman-group1-sha1",
"diffie-hellman-group14-sha1"
],
```
without any results.

Can somebody help me please?
Thanks

Tom

**Comment by drakkan** [comment]
Hi,

the settings you modified are for the server side of SFTPGo, please reset them to the defaults.
If I understand correctly your question you are using an old SFTP server as storage backend in SFTPGo. Please try the development version it should work
